File: LAB3/app/src/main/java/com/example/tipper/BmiActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class BmiActivity extends AppCompatActivity {

    EditText height, weight;
    TextView result;
    Button calculate, btnBack;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmi);

        height = (EditText) findViewById(R.id.height);
        weight = (EditText) findViewById(R.id.weight);
        result = (TextView) findViewById(R.id.result);
        calculate = (Button) findViewById(R.id.calculate);

        calculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                calculateBMI();
            }
        });

        btnBack = (Button) findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(BmiActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }

    private void calculateBMI() {
        String heightStr = height.getText().toString();
        String weightStr = weight.getText().toString();

        if (heightStr != null && !"".equals(heightStr) && weightStr != null && !"".equals(weightStr)) {
            float heightValue = Float.parseFloat(heightStr) / 100;
            float weightValue = Float.parseFloat(weightStr);

            float bmi = weightValue / (heightValue * heightValue);

            displayBMI(bmi);
        }
    }

    private void displayBMI(float bmi) {
        String bmiLabel = "";

        if(Float.compare(bmi, 15f) <= 0) {
            bmiLabel = "wygłodzenie";
        } else if (Float.compare(bmi, 15f) > 0 && Float.compare(bmi, 16f) <= 0) {
            bmiLabel = "wychudzenie";
        } else if (Float.compare(bmi, 16f) > 0 && Float.compare(bmi, 18.5f) <= 0) {
            bmiLabel = "niedowaga";
        } else if (Float.compare(bmi, 18.5f) > 0 && Float.compare(bmi, 25f) <= 0) {
            bmiLabel = "pożądana masa ciała";
        } else if (Float.compare(bmi, 25f) > 0 && Float.compare(bmi, 30f) <= 0) {
            bmiLabel = "nadwaga";
        } else if (Float.compare(bmi, 30f) > 0 && Float.compare(bmi, 35f) <= 0) {
            bmiLabel = "otyłość I stopnia";
        } else if (Float.compare(bmi, 35f) > 0 && Float.compare(bmi, 40f) <= 0) {
            bmiLabel = "otyłość II stopnia";
        } else {
            bmiLabel = "otyłość III stopnia";
        }

        bmiLabel = bmi + "\n" + bmiLabel;
        result.setText(bmiLabel);
    }
}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/BmiActivity.kt
package com.example.myapplicationpamo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView


class BmiActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmi)

        val calculate = findViewById<Button>(R.id.calculate);
        val weight = findViewById<EditText>(R.id.weight);
        val height = findViewById<EditText>(R.id.height);
        val result = findViewById<TextView>(R.id.result);



        calculate.setOnClickListener() {
            val input_weight: String = weight.text.toString()
            val value: Double = input_weight.toDouble()


            val input_height: String = height.text.toString()
            val value1: Double = input_height.toDouble() / 100

            val bmiresult = calculateBMI(value,value1)



            if (bmiresult < 18.5) {
                result.text =
                    "%.2f".format(bmiresult) + " wygłodzenie"
            } else if (bmiresult <= 24.9) {
                result.text =
                    "%.2f".format(bmiresult) + " pożądana masa ciała"
            } else {
                result.text =
                    "%.2f".format(bmiresult) + " nadwaga"
            }
        }




    }
    private fun calculateBMI(v1: Double, v2: Double): Double {
        val bmiresult: Double = v1 / (v2 * v2)
        return bmiresult

    }
}